### PR TITLE
[MNG-7895] Support ${project.basedir} in file profile activation

### PR DESCRIPTION
**Pull request by slawekjaranowski**
https://issues.apache.org/jira/browse/MNG-7895

ITs - https://github.com/apache/maven-integration-testing/pull/299
 
---

Following this checklist to help us incorporate your
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] SUMMARY`,
       where you replace `MNG-XXX` and `SUMMARY` with the appropriate JIRA issue.
 - x ] Also format the first line of the commit message like `[MNG-XXX] SUMMARY`.
       Best practice is to use the JIRA issue title in both the pull request title and in the first line of the commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [x] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
Jenkins use master branch for ITs ...
Connected ITs fix - https://github.com/apache/maven-integration-testing/pull/299

File: api/maven-api-model/src/main/mdo/maven.mdo [hunk 1]
```diff
@@ -3007,7 +3007,7 @@
         is the location of a file that needs to exist, and if it doesn't, the profile will be
         activated. On the other hand, {@code exists} will test for the existence of the file and if it is
         there, the profile will be activated.<br>
-        Variable interpolation for these file specifications is limited to {@code ${basedir}},
+        Variable interpolation for these file specifications is limited to {@code ${project.basedir}},
```
Comment (gnodet) [inline api/maven-api-model/src/main/mdo/maven.mdo:3010]: Would it make sense to also support `session.rootDirectory` ?

Comment (slawekjaranowski) [reply api/maven-api-model/src/main/mdo/maven.mdo:3010]: We can add - currently we have implementation for `rootDirectory` - but I would like change to `project.rootDirectory`
https://issues.apache.org/jira/browse/MNG-7897

I would like to cherry-pick this for 3.9.5 - so only one change in one PR

**Comment by cstamas** [comment]
Hm, now by reading this PR and seeing several places where comment explains why NOT `project.basedir`... it makes me worried, but is LGTM

**Comment by slawekjaranowski** [comment]
> Hm, now by reading this PR and seeing several places where comment explains why NOT `project.basedir`... it makes me worried, but is LGTM

But in other place we deprecated short properties like `basedir` so it can consusing users also. 
We should be consistent.

**Comment by gnodet** [comment]
> > Hm, now by reading this PR and seeing several places where comment explains why NOT `project.basedir`... it makes me worried, but is LGTM
> 
> But in other place we deprecated short properties like `basedir` so it can consusing users also. We should be consistent.

Then should we deprecate the `project.basedir` ? i.e. add a warning when non prefixed properties are used ?

**Comment by slawekjaranowski** [comment]
> > > Hm, now by reading this PR and seeing several places where comment explains why NOT `project.basedir`... it makes me worried, but is LGTM
> > 
> > 
> > But in other place we deprecated short properties like `basedir` so it can consusing users also. We should be consistent.
> 
> Then should we deprecate the `basedir` ? i.e. add a warning when non prefixed properties are used ?

I can add here a warning about using deprecated `basedir` in profile activation ....
We can also check other places where `basedir` is used .. but should be done in separate PR

**Review by gnodet — APPROVED** [review]

**Review by cstamas — APPROVED** [review]

**Pull request merged**